# Flights Comparison of Alaska and AM West Airlines

For this project, I focused on a simple on-time flight comparison analysis between the Alaska and AM West Airlines. The comparision includes destination flights to Los Angeles, Phoenix, San Diego, San Francisco, and Seattle. The intital steps involved importing pandas and numpy as well as importing the data tables containing the number of on-time and delayed flights. 

In [1]:
#imports pandas and Numpy 
import pandas as pd
import numpy as np
from pandas import DataFrame, Series

In [2]:
#Assigns project1.csv path to "Location"
Location = r'C:\Users\rev0c\OneDrive\Documents\SPS\IS 362 - Data Acquisition and Management\project1.csv'

In [3]:
#Reads data from "Location" and assigns to "df"
df = pd.read_csv(Location)

The initial tables not only includes the airlines and cities but the number of on-time and delayed flights.

In [6]:
#Displays 'project1.csv'
df

airline           city  on time  delayed
0  ALASKA    Los Angeles      497       62
1  ALASKA        Phoenix      221       12
2  ALASKA      San Diego      212       20
3  ALASKA  San Francisco      503      102
4  ALASKA        Seattle     1841      305
5  AMWEST    Los Angeles      694      117
6  AMWEST        Phoenix     4840      415
7  AMWEST      San Diego      393       65
8  AMWEST  San Francisco      320      129
9  AMWEST        Seattle      201       61

I created a copy of the data into a dataframe named "df1."

In [9]:
#Copies data to 'df1'
df1 = df.copy()

To add the additional columns, I used "df1['total # of flights']" and "df1['on time flights (%)']" since I could not find a simpler solution. I will modify this once I get more comfortable with pandas. 

In [10]:
#Add "on time(%)" and "total # of flights" columns
df1['total # of flights'] = ''
df1['on time flight (%)'] = ''

Below is the table with the new "total # of flights" and "on time flights (%)" columns.

In [11]:
#Shows updated df1 table
df1

airline           city  on time  delayed total # of flights  \
0  ALASKA    Los Angeles      497       62                      
1  ALASKA        Phoenix      221       12                      
2  ALASKA      San Diego      212       20                      
3  ALASKA  San Francisco      503      102                      
4  ALASKA        Seattle     1841      305                      
5  AMWEST    Los Angeles      694      117                      
6  AMWEST        Phoenix     4840      415                      
7  AMWEST      San Diego      393       65                      
8  AMWEST  San Francisco      320      129                      
9  AMWEST        Seattle      201       61                      

  on time flight (%)  
0                     
1                     
2                     
3                     
4                     
5                     
6                     
7                     
8                     
9

I used the sum function to identify the total number of flights for each airline and city. The results are shown below.

In [13]:
#Calculates the total number of flights for each airline and destination 
df1['total # of flights'] = df1.sum(axis=1)

In [14]:
df1

airline           city  on time  delayed  total # of flights  \
0  ALASKA    Los Angeles      497       62                 559   
1  ALASKA        Phoenix      221       12                 233   
2  ALASKA      San Diego      212       20                 232   
3  ALASKA  San Francisco      503      102                 605   
4  ALASKA        Seattle     1841      305                2146   
5  AMWEST    Los Angeles      694      117                 811   
6  AMWEST        Phoenix     4840      415                5255   
7  AMWEST      San Diego      393       65                 458   
8  AMWEST  San Francisco      320      129                 449   
9  AMWEST        Seattle      201       61                 262   

  on time flight (%)  
0                     
1                     
2                     
3                     
4                     
5                     
6                     
7                     
8                     
9

Having found the total number of flights, I was able to calculate the percentage of on-time flights. This required  simply dividing on-time flights by total number of flights. The results are shown below.

In [15]:
#Calculates the on time flight percentage for each airline and destination
df1['on time flight (%)'] = (df1['on time']/df1['total # of flights']) *100

In [16]:
# Displays the updated table containing the on time flight percentage
df1

airline           city  on time  delayed  total # of flights  \
0  ALASKA    Los Angeles      497       62                 559   
1  ALASKA        Phoenix      221       12                 233   
2  ALASKA      San Diego      212       20                 232   
3  ALASKA  San Francisco      503      102                 605   
4  ALASKA        Seattle     1841      305                2146   
5  AMWEST    Los Angeles      694      117                 811   
6  AMWEST        Phoenix     4840      415                5255   
7  AMWEST      San Diego      393       65                 458   
8  AMWEST  San Francisco      320      129                 449   
9  AMWEST        Seattle      201       61                 262   

   on time flight (%)  
0           88.908766  
1           94.849785  
2           91.379310  
3           83.140496  
4           85.787512  
5           85.573366  
6           92.102759  
7           85.807860  
8           71.269488  
9           76.717557

The final step involved organizing the information for easier comparsion by city. The results are shown below.

In [17]:
#Sorts and updates by city
df1 = df1.sort_values(by='city',ascending=0)

In [18]:
#Displays 'df1' table sorted by city
df1

airline           city  on time  delayed  total # of flights  \
4  ALASKA        Seattle     1841      305                2146   
9  AMWEST        Seattle      201       61                 262   
3  ALASKA  San Francisco      503      102                 605   
8  AMWEST  San Francisco      320      129                 449   
2  ALASKA      San Diego      212       20                 232   
7  AMWEST      San Diego      393       65                 458   
1  ALASKA        Phoenix      221       12                 233   
6  AMWEST        Phoenix     4840      415                5255   
0  ALASKA    Los Angeles      497       62                 559   
5  AMWEST    Los Angeles      694      117                 811   

   on time flight (%)  
4           85.787512  
9           76.717557  
3           83.140496  
8           71.269488  
2           91.379310  
7           85.807860  
1           94.849785  
6           92.102759  
0           88.908766  
5           85.573366

#### Conclusion:
Comparison of the on-time flight percentages between Alaska and AM West showed Alaska having the better overall percentage. The lowest on-time flight percentages for Alaska (83.1405%) and AM West (71.2695%)was in San Francisco. The highest on-time flight performance for both (Alaska: 91.379310% and AM West: 92.102759%) was in Phoenix. Phoenix is also the AM West has the highest number of flights. Further analysis would be performed to determine the reasoning behind the discrepancies.      